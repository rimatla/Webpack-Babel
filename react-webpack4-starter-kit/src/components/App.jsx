import React from 'react'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import AsyncRoute from '../AsyncRoute'

import './App.css'

// code splitting
const AsyncHome = props => (
  <AsyncRoute props={props} loadingPromise={import('./Home')} />
)

const AsyncHello = props => (
  <AsyncRoute props={props} loadingPromise={import('./Hello')} />
)

const AsyncWorld = props => (
  <AsyncRoute props={props} loadingPromise={import('./World')} />
)

const AsyncNotFound = props => (
  <AsyncRoute props={props} loadingPromise={import('./NotFound')} />
)

const App = () => (
  <div className="app">
    <BrowserRouter>
      <Switch>
        <Route exact path="/" component={AsyncHome} />
        <Route path="/hello" component={AsyncHello} />
        <Route path="/world" component={AsyncWorld} />
        <Route component={AsyncNotFound} />
      </Switch>
    </BrowserRouter>
  </div>
)

export default App
