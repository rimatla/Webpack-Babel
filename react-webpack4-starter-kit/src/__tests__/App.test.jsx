import React from 'react'
import { shallow } from 'enzyme'
import App from '../components/App'

test('renders correctly', () => {
  const component = shallow(<App />)
  expect(component).toMatchSnapshot()
})
