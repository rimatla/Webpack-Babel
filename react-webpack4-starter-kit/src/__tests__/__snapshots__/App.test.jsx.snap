// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/hello"
      />
      <Route
        component={[Function]}
        path="/world"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;
